core/templates/services/state-interaction-stats.service.ts

```diff
@@ -141,9 +144,10 @@ export class StateInteractionStatsService {
                 undefined,
             }),
           })),
-        });
-    }
-    return this.cachedStats;
+        };
+        this.cachedStats.set(state.name, stateInteractionStats);
+        return stateInteractionStats;
+      });
```

Comment:
Why don't you cache the promises in this code?

Reply:
I think the good think about caching the promises is that when the promise is not yet completed it can still be used in the second request and after it finishes it is resolved on both the places.

Reply:
So please bring it back.